app/lib/actions/pulp3/capsule_content/sync.rb

```diff
@@ -0,0 +1,26 @@
+module Actions
+  module Pulp3
+    module CapsuleContent
+      class Sync < Pulp3::AbstractAsyncTask
+        def plan(repository, smart_proxy, options = {})
+          sequence do
+            plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id, :options => options)
+            plan_action(GenerateMetadata, repository, smart_proxy, options)
+          end
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find(input[:repository_id])
+          output[:pulp_tasks] = repo.backend_service(::SmartProxy.unscoped.find(input[:smart_proxy_id])).sync_mirror
+        end
```

Comment:
I don't know how often full capsule sync is run but we would be syncing all repositories on the capsule on every refresh with this?

Reply:
Actually only syncable repos are synced. This was the case even in the prior implementation -> 

https://github.com/Katello/katello/blob/master/app/lib/actions/katello/capsule_content/sync.rb#L43-L60

Its the same set here also.